client/src/components/NemPriceGraph.vue

```diff
@@ -66,137 +65,57 @@
-        )
-      ])
-      .then(
-        axios.spread(res1 => {
-          // var x = []
-          // var y = []
-          self.loading = 1
-          this.drawCandleChart(res1.data, '#nempricegraph')
-        })
-      )
-      .catch(error => {
-        console.log(error)
-      })
-  },
-  methods: {
-    drawCandleChart(data, elmnt) {
+  computed: {
+    historicalHourlyGraph() {
       let graphData = []
-      data.Data.forEach((item, index) => {
-        let graphDataItem = {}
-        graphDataItem.y = []
-        graphDataItem.x = new Date(item['time'] * 1000)
-        graphDataItem.y[0] = item['open'] // parseFloat(item['open']).toFixed(4)
-        graphDataItem.y[1] = item['high'] // parseFloat(item['high']).toFixed(4)
-        graphDataItem.y[2] = item['low'] // parseFloat(item['low']).toFixed(4)
-        graphDataItem.y[3] = item['close'] // parseFloat(item['close']).toFixed(4)
-        graphData.push(graphDataItem)
-      })
-      let options = {
-        chart: {
-          height: 200,
-          type: 'area',
-          foreColor: '#999',
-          toolbar: {
-            show: false
-            // tools: {
-            //   download: false,
-            //   selection: true,
-            //   zoom: false,
-            //   zoomin: true,
-            //   zoomout: true,
-            //   pan: '<i class="ico-arrow-move" style="font-size: 24px;"></i>',
-            //   reset: '<i class="ico-ios-refresh-outline" style="font-size: 22px;"></i>',
-            //   customIcons: []
-            // }
-          }
-        },
-        stroke: {
-          show: true,
-          curve: 'smooth',
-          lineCap: 'butt',
-          colors: ['#0998a6'],
-          width: 2,
-          dashArray: 0
-        },
-        dataLabels: {
-          enabled: false
-        },
-        colors: ['#1eaaa6'],
-        fill: {
-          gradient: {
-            enabled: true,
-            opacityFrom: 1,
-            opacityTo: 0.7
-          }
-        },
-        plotOptions: {
-          candlestick: {
-            colors: {
-              upward: '#0998a6',
-              downward: '#f7a800'
-            },
-            wick: {
-              useFillColor: true
-            }
-          }
-        },
-        series: [
-          {
-            name: 'Price',
-            data: graphData
-          }
-        ],
-        title: {
-          text: '',
-          align: 'left',
-          display: false
-        },
-        xaxis: {
-          type: 'datetime',
-          axisBorder: {
-            show: false,
-            color: '#0998a6'
-          }
-        },
-        yaxis: {
-          tooltip: {
-            enabled: true
-          },
-          axisBorder: {
-            show: false,
-            color: '#0998a6'
-          }
-        },
-        tooltip: {
-          enabled: true
-        }
+      if (this.marketData.historicalHourlyGraph.Data) {
+        this.marketData.historicalHourlyGraph.Data.forEach((item, index) => {
```

Comment:
I think it is better to move this part to the Store

Reply:
will move to store :)